barril/units/unit_database.py

```diff
@@ -0,0 +1,1311 @@
+from __future__ import absolute_import, division, unicode_literals
+
+import attr
+import copy
+import math
+import traceback
+
+import six
+from six.moves import zip  # @UnresolvedImport
+
+from barril.foundation.singleton import Singleton
```

Comment:
I think that our helpers modules should be prefixed by `_` (i.e.: `_foundation`) to make it explicit that they should not be considered part of the public API.